.github/workflows/windows-ci.yml

```diff
@@ -21,7 +21,7 @@ jobs:
       run: dotnet restore
 
     - name: Build
-      run: dotnet build --configuration Release --no-restore
+      run: dotnet build --configuration Release --no-restore -p:BuildNumber=$GITHUB_RUN_NUMBER
```

Comment:
instead of modifying all CI builds, we can modify the version generation part more robust, so it can work even if buildnumber is not specified, by falling back to 0 (as before)